learning_assistant/utils.py

```diff
@@ -12,7 +12,42 @@
 log = logging.getLogger(__name__)
 
 
-def get_chat_response(message_list):
+def _estimated_message_tokens(message):
+    """
+    Estimates how many tokens are in a given message.
+    """
+    chars_per_token = 3.5
+    json_padding = 8
+
+    return int((len(message) - message.count(' ')) / chars_per_token) + json_padding
+
+
+def get_reduced_message_list(system_list, message_list):
```

Comment:
This was largely taken from https://github.com/2uinc/ai-chatbot-backend/blob/74faef96cc3732e1369d3ad5adeae9cdd79d0bc8/xpert/pipelines/utils.py#L173. I modified a few things within the while loop (like moving a break and adding it to the outer conditional) and changing the append + reverse to an insert for the `new_message_list`